src/lib.rs

```diff
@@ -48,43 +53,17 @@ fn run_game_loop(solution: &String, user_facing_message: &mut String, input: cha
 }
 
 pub fn run(config: Config) -> Result<(), Box<dyn Error>> {
-    let solution = fs::read_to_string(config.file_path)?;
-    let mut user_facing_message: String = "_".repeat(solution.len());
+    let mut user_facing_message: String = "_".repeat(config.solution.len());
 
     println!("{}", user_facing_message);
 
-    while (run_game_loop(&solution, &mut user_facing_message, take_guess())) {}
+    while (run_game_loop(&config.solution, &mut user_facing_message, take_guess())) {}
 
     Ok(())
 }
 
 pub struct Config {
-    pub guess: String,
-    pub file_path: String,
-    pub ignore_case: bool,
-}
-
-impl Config {
-    pub fn build(mut args: impl Iterator<Item = String>) -> Result<Config, &'static str> {
```

Comment:
I noticed that here we move away from the pattern of implementing a `build` method for our `Config` struct and now instead you added the `take_config` function.

Reply:
👍🏻 see [this comment](https://github.com/h0uter/galgo/pull/2#discussion_r1703270373)